src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectMethodExecutor.cs

```diff
@@ -67,6 +71,18 @@ public object Execute(object target, object[] parameters)
             return _executor(target, parameters);
         }
 
+        public object GetDefaultValueForParameter(int index)
+        {
+            if (index < 0 || index > ActionParameters.Length-1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+
+            EnsureParameterDefaultValues();
```

Comment:
I see the benefit of making it lazy because we don't compute default values always. We do it only if there aren't any action parameters for the corresponding method parameters which may not happen very often. We can save the object array allocation per controller action method and also querying the custom attributes  by making it lazy.